### PR TITLE
NR-401130: Added version support of Azure SQL in nri-mssql.

### PR DESCRIPTION
**Pull request by satya-nr**
What it does ?

Added support for Microsoft SQL Azure (RTM) - 12.0.2000.8 in the nri-mssql. The current nri-mssql versions support standard SQL from major version numbers 14 to 16. This PR extends support to include versions 12 to 14 specifically for Azure SQL, with the Azure cloud currently using version 12. Support up to version 16 has been included for future compatibility."

Testing:

The test cases for Azure SQL version were executed with the scenarios listed below, and all tests passed successfully.
1) go test -v -run TestCheckSQLServerVersionforAzure

### CODE REVIEW

File: src/queryanalysis/validation/sql_server_version.go [hunk 1]
```diff
@@ -11,10 +12,11 @@ import (
 )
 
 const (
-	versionRegexPattern      = `\b(\d+\.\d+\.\d+)\b`
-	getSQLServerVersionQuery = "SELECT @@VERSION"
-	lastSupportedVersion     = 16
-	firstSupportedVersion    = 14
+	versionRegexPattern        = `\b(\d+\.\d+\.\d+)\b`
+	getSQLServerVersionQuery   = "SELECT @@VERSION"
+	lastSupportedVersion       = 16
```
Comment (rajrohanyadav) [inline src/queryanalysis/validation/sql_server_version.go:17]: Do we need another variable for maintaining last supported version for azure and non-azure?

File: src/queryanalysis/validation/sql_server_version.go [hunk 2]
```diff
@@ -11,10 +12,11 @@ import (
...
+	firstSupportedVersion      = 14
+	azureFirstSupportedVersion = 12
```
Comment (rajrohanyadav) [inline src/queryanalysis/validation/sql_server_version.go:19]: I think it makes sense to add a comment stating what this variable refers to. e.g. the kind of deployment.

File: src/queryanalysis/validation/sql_server_version_test.go [hunk 3]
```diff
@@ -22,6 +22,32 @@ func TestCheckSqlServerVersion_SupportedVersion(t *testing.T) {
...
+	defer sqlConnection.Connection.Close()
+
+	// Mocking an unsupported Azure SQL Server version response
+	mock.ExpectQuery(regexp.QuoteMeta(getSQLServerVersionQuery)).
+		WillReturnRows(sqlmock.NewRows([]string{"@@VERSION"}).AddRow("Microsoft SQL Azure (RTM) - 11.0.2000.7"))
+
+	result, _ := checkSQLServerVersion(sqlConnection)
+	assert.False(t, result)
+	assert.NoError(t, mock.ExpectationsWereMet())
+}
```
Comment (rajrohanyadav) [inline src/queryanalysis/validation/sql_server_version_test.go:49]: These two tests have almost the same code except the assertions. Can we use table driven tests? It will help us in adding some negative tests as well (because that is my second suggestion, please add some negative tests)

**Review by rajrohanyadav — COMMENTED** [review]
Thanks, added a few comments.

File: src/queryanalysis/validation/sql_server_version.go (hunk 1 above)
Comment (satya-nr) [reply src/queryanalysis/validation/sql_server_version.go]: Yes i added because to differentiate standard SQL and Azure SQL versions and also i don't want break existed code logic, that is the reason added another variable.

Comment (rajrohanyadav) [reply src/queryanalysis/validation/sql_server_version.go]: No, I meant, let's add another one for `lastSupportedVersion` because right now it is common for both formats. Does it make sense that we use a different variable for this?

File: src/queryanalysis/validation/sql_server_version.go [hunk 4]
```diff
@@ -15,6 +16,9 @@ const (
 	getSQLServerVersionQuery = "SELECT @@VERSION"
 	lastSupportedVersion     = 16
 	firstSupportedVersion    = 14
+	// Represents the first supported version for Azure SQL Server in the cloud.
```
Comment (rajrohanyadav) [inline src/queryanalysis/validation/sql_server_version.go:19]: nit: I think you forgot to update the comment.

File: src/queryanalysis/validation/sql_server_version_test.go [hunk 5]
```diff
@@ -22,6 +22,33 @@ func TestCheckSqlServerVersion_SupportedVersion(t *testing.T) {
...
 }
 
+func TestCheckSQLServerVersionforAzure(t *testing.T) {
+	tests := []struct {
+		name     string
+		version  string
+		expected bool
+	}{
+		{"AzureSupportedVersion", "Microsoft SQL Azure (RTM) - 12.0.2000.8", true},
+		{"AzureUnsupportedVersion", "Microsoft SQL Azure (RTM) - 11.0.2000.7", false},
```
Comment (rajrohanyadav) [inline src/queryanalysis/validation/sql_server_version_test.go:32]: Does this also need tests when `azure` is not in the version?

**Review by rajrohanyadav — APPROVED** [review]
LGTM

File: src/queryanalysis/validation/sql_server_version_test.go (hunk 5 above)
Comment (satya-nr) [reply src/queryanalysis/validation/sql_server_version_test.go]: Yes make sense i added another case for it and also added negative test case for highest unsupported version.

File: src/queryanalysis/validation/sql_server_version.go (hunk 4 above)
Comment (satya-nr) [reply src/queryanalysis/validation/sql_server_version.go]: Updated comment for it.

**Review by rajrohanyadav — APPROVED** [review]
